src/main/java/javax/validation/Configuration.java

```diff
@@ -127,6 +129,19 @@
 	T clockProvider(ClockProvider clockProvider);
 
 	/**
+	 * Adds a value extractor. Has priority over any extractors for the same
+	 * type and type parameter detected through the service loader or given in
+	 * the XML configuration.
+	 *
+	 * @param extractor value extractor implementation
+	 * @return {@code this} following the chaining method pattern.
+	 * @throws IllegalArgumentException If more than one extractor for the same
```

Comment:
I'm also not completely sure about this. But I dislike throwing an exception to be honest. So using the last one looks like a better option to me.